src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -927,6 +927,17 @@ public BoundStatement ThrowNull()
             return Throw(Null(Compilation.GetWellKnownType(Microsoft.CodeAnalysis.WellKnownType.System_Exception)));
         }
 
+        public BoundExpression ThrowNullExpression(TypeSymbol type)
+        {
+            return new BoundThrowExpression(Syntax, Null(Compilation.GetWellKnownType(CodeAnalysis.WellKnownType.System_Exception)), type)
+                { WasCompilerGenerated = true };
```

Comment:
I think, It would be better to throw Object. We will end up with IL for "throw null" anyways, so using System_Exception here just makes this code dependent on presence of Exception type, which it does not need to function.